frontend/lib/resolution/return-type-inference.cpp

```diff
@@ -100,13 +103,27 @@ const CompositeType* helpGetTypeForDecl(Context* context,
           parentClassType = ct->basicClassType();
         }
       }
+
       if (qt.isType() && parentClassType != nullptr) {
+        // It's a valid parent class; is it the only one? (error otherwise).
+        if (lastParentClass) {
+          reportInvalidMultipleInheritance(context, c, lastParentClass, parentExpr);
```

Comment:
If you inherit from 3 parent classes won't you print the error multiple times? I would imagine the error is better if it's only printed once? Seems you need to break out of the loop here or else move this check outside of the loop (by tracking the firstParentClass and the lastParentClass).